### PR TITLE
Make literal integer font lock face customizable

### LINKED ISSUE #29: Hexadecimal digits prefix highlights as keyword
image

It can be reproduced using the following `init.el` with `emacs -Q -l init.el`.

```elisp
(require 'package)
(setq package-archives
      '(("gnu"   . "http://mirrors.tuna.tsinghua.edu.cn/elpa/gnu/")
        ("melpa" . "http://mirrors.tuna.tsinghua.edu.cn/elpa/melpa/")))

(package-initialize)

;; Bootstrap `use-package'
(unless (package-installed-p 'use-package)
  (package-refresh-contents)
  (package-install 'use-package))
(eval-when-compile
  (require 'use-package))

(use-package modern-cpp-font-lock
  :ensure t
  :hook (c++-mode . modern-c++-font-lock-mode))

(provide 'init)
```

emacs --version

``` text
GNU Emacs 28.0.50
Copyright (C) 2020 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of GNU Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.
```

cc-mode version

```
;; Version: 5.33.1
```

After reading the source, it seems like a feature but a bug.

https://github.com/ludwigpacifici/modern-cpp-font-lock/blob/master/modern-cpp-font-lock.el#L186

### PR DESCRIPTION
**Pull request by condy0919**
Fixes #29

### CODE REVIEW

**Comment by condy0919** [comment]
`*-prefix-face`, `*-infix-face`, and `*-suffix-face` are bad names, wordings are welcome.

**Comment by condy0919** [comment]
@ludwigpacifici

File: modern-cpp-font-lock.el [hunk 1]
```diff
@@ -163,6 +163,65 @@ http://en.cppreference.com/w/cpp/language/integer_literal"
   :type 'boolean
   :group 'modern-c++-font-lock)
 
+;; Binary literal integers
+(defcustom modern-c++-literal-binary-integer-prefix-face 'font-lock-keyword-face
```
Comment (ludwigpacifici) [inline modern-cpp-font-lock.el]: I find the wording fine, maybe a bit too long. You can shorten it a bit by dropping `integer`, for example: `modern-c++-literal-binary-prefix-face` (and apply it to other new variables).

Why do you find the wording bad?

**Review by ludwigpacifici — CHANGES_REQUESTED** [review]
Thank you for the PR! Overall, it looks good.

One renaming suggestion. Also, can you add some documentation in [here](modern-c++-literal-binary-integer-prefix-face) to show how to customize it.

File: modern-cpp-font-lock.el (hunk 1 above)
Comment (condy0919) [reply modern-cpp-font-lock.el]: `prefix`, `infix` and `suffix` are not semantic

File: README.md [hunk 2]
```diff
@@ -81,6 +81,20 @@ Set to `t` (default value) to enable the following options or otherwise to `nil`
  * `modern-c++-literal-null-pointer` - `nullptr`
  * `modern-c++-literal-string` - `R"xyz()")xyz"`, `L"hello\ngoodbye"`, `"abcd"`, [etc](http://en.cppreference.com/w/cpp/language/string_literal).
 
+Configure the following lists to get proper literal integer font locking:
```
Comment (ludwigpacifici) [inline README.md]: I would rephrase it "Configure the following list to customize font locking for literal integers"

File: modern-cpp-font-lock.el [hunk 3]
```diff
@@ -163,6 +163,65 @@ http://en.cppreference.com/w/cpp/language/integer_literal"
   :type 'boolean
   :group 'modern-c++-font-lock)
 
+;; Binary literal integers
```
Comment (ludwigpacifici) [inline modern-cpp-font-lock.el]: Do you mind to remove this comment? It does not more information than your below code.

File: modern-cpp-font-lock.el [hunk 4]
```diff
@@ -163,6 +163,65 @@ http://en.cppreference.com/w/cpp/language/integer_literal"
...
+  "Face for displaying binary literal integer infix."
+  :type 'symbol
+  :group 'modern-c++-font-lock)
+
+(defcustom modern-c++-literal-binary-suffix-face 'font-lock-keyword-face
+  "Face for displaying binary literal integer suffix."
+  :type 'symbol
+  :group 'modern-c++-font-lock)
+
+;; Octal literal integers
```
Comment (ludwigpacifici) [inline modern-cpp-font-lock.el]: Do you mind to remove this comment? Same reason as above.

File: modern-cpp-font-lock.el [hunk 5]
```diff
@@ -163,6 +163,65 @@ http://en.cppreference.com/w/cpp/language/integer_literal"
...
+  "Face for displaying hexadecimal literal integer infix."
+  :type 'symbol
+  :group 'modern-c++-font-lock)
+
+(defcustom modern-c++-literal-hex-suffix-face 'font-lock-keyword-face
+  "Face for displaying hexadecimal literal integer suffix."
+  :type 'symbol
+  :group 'modern-c++-font-lock)
+
+;; Decimal literal integers
```
Comment (ludwigpacifici) [inline modern-cpp-font-lock.el]: Do you mind to remove this comment? Same reason as above.

**Review by ludwigpacifici — CHANGES_REQUESTED** [review]
Thanks @condy0919. A few more comments and then it is good do merge.

**Comment by ludwigpacifici** [comment]
Also, if you wish, feel free to add your name [here](https://github.com/ludwigpacifici/modern-cpp-font-lock/blob/master/CONTRIBUTING.md).

**Comment by condy0919** [comment]
> Also, if you wish, feel free to add your name [here](https://github.com/ludwigpacifici/modern-cpp-font-lock/blob/master/CONTRIBUTING.md).

Thanks! Added

File: README.md (hunk 2 above)
Comment (condy0919) [reply README.md]: Thanks for better wording!
